backend/src/common/plugins/auth/auth-jwt.plugin.ts

```diff
@@ -0,0 +1,60 @@
+import fp from 'fastify-plugin';
+import { HttpCode, HttpError, HttpHeader } from 'shared';
+
+import { userService } from '~/bundles/users/users.js';
+import { tokenService } from '~/common/services/services.js';
+
+import { ErrorMessage, Hook } from './enums/enums.js';
+
+type Options = {
+    routesWhiteList: string[];
+};
+
+const authenticateJWT = fp<Options>((fastify, options, done) => {
+    fastify.decorateRequest('user', null);
+
+    fastify.addHook(Hook.PRE_HANDLER, async (request) => {
+        const isRouteInWhiteList = options.routesWhiteList.includes(
+            request.url,
+        );
```

Comment:
also check for method in whitelist because each route can have multiple methods

Reply:
done